memcache/memcache.go

```diff
@@ -89,7 +89,7 @@ func legalKey(key string) bool {
 		return false
 	}
 	for i := 0; i < len(key); i++ {
-		if key[i] <= ' ' || key[i] > 0x7e {
+		if key[i] <= ' ' || key[i] == 0x7f {
```

Comment:
The whole point is for us to support non ASCII characters,for ventures that use non-latin letters, isn't it? 
For ex, thai characters: (0E00 - 0E7F) http://www.unicode.org/charts/PDF/U0E00.pdf